force-app/main/default/classes/DMLOperations.cls

```diff
@@ -129,10 +155,18 @@ public with sharing class DMLOperations {
      */
     public static void upsertOpportunityList(List<Opportunity> opportunities) {
         // Loop through each Opportunity in the list
+        Date todaysDate = Date.today().addMonths(3);
```

Comment:
Great job handling the date calculation outside the loop to avoid re-processing the same logic over and over.